src/app/auth/authentication.service.ts

```diff
@@ -32,13 +34,17 @@ export class AuthenticationService {
     @Inject(AUTH_API_URL) apiUrl: string,
     @Inject(SSO_API_URL) ssoUrl: string,
     @Inject(REALM) realm: string,
-    private http: Http
+    private http: Http,
+    private logger: Logger
   ) {
     this.apiUrl = apiUrl;
     this.ssoUrl = ssoUrl;
     this.realm = realm;
     this.openShiftToken = this.createFederatedToken(this.openshift, (response: Response) => response.json() as Token);
     this.gitHubToken = this.createFederatedToken(this.github, (response: Response) => response.json() as Token);
+    if (this.getToken() != null) {
+      this.headers.set('Authorization', 'Bearer ' + this.getToken());
```

Comment:
Please do not set the headers in here - we have a shared httpclient that should be used as a provider globally.